pkg/apis/openebs.io/scp/v1alpha1/capabilties.go

```diff
@@ -0,0 +1,125 @@
+
+	// IO connectivity capabilities
+	// +nullable
+	IOConnectivity *IOConnectivityCapabilities `json:"IOConnectivity,omitempty"`
+
+	// IO performance capabilities
+	IOPerformance *IOPerformanceCapabilities `json:"IOPerformance"`
+
+	// DataProtection capabilities
+	// +nullable
+	DataProtection *DataProtectionCapabilities `json:"dataProtection,omitempty"`
+}
+
+// DataStorageCapabilities defines data storage capabilities for a pool or volume
+type DataStorageCapabilities struct {
+	//AccessModes contains the actual access modes
+	AccessModes []AccessMode `json:"accessModes"`
+
+	// ProvisioningPolicy defines provisioning policy type
+	ProvisioningPolicy []string `json:"provisioningPolicy"`
+
+	// MultiPathing to be supported or not.
+	// +nullable
+	MultiPathing []string `json:"multiPathing,omitempty"`
+
+	// Compression to be supported or not, if yes the algorithm
+	//	+nullable
+	Compression []string `json:"compression,omitempty"`
+
+	// Deduplication to be supported or not.
+	// +nullable
+	Deduplication []string `json:"deduplication,omitempty"`
+}
+
+// AccessMode of the pool or volume
+type AccessMode string
+
+const (
+	// ReadWriteOnce can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce AccessMode = "ReadWriteOnce"
+
+	// ReadOnlyMany can be mounted in read-only mode to many hosts
+	ReadOnlyMany AccessMode = "ReadOnlyMany"
+
+	// ReadWriteMany can be mounted in read/write mode to many hosts
+	ReadWriteMany AccessMode = "ReadWriteMany"
+)
+
+
+// DataSecurityCapabilities defines media encryption algorithms and data sanitization policy.
+type DataSecurityCapabilities struct {
+
+	// MediaEncryption to be supported or not, if yes the algorithms
+	// +nullable
+	MediaEncryption []string `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy to be supported, viz Clear, CryptographicErase
+	// +nullable
+	DataSanitizationPolicy []string `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// IOConnectivityCapabilities defines access protocols to be supported by the pool or volume
+type IOConnectivityCapabilities struct {
+	// AccessProtocols to be supported, viz NVMe, NVMeOverFabrics
+	// +nullable
+	AccessProtocols []string `json:"accessProtocols,omitempty"`
+}
+
+// IOPerformanceCapabilities defines IO performance capabilities for the pool or volume.
+type IOPerformanceCapabilities struct {
+	// AverageIOOperationLatencyMicroseconds to be supported or not.
+	// +nullable
+	AverageIOOperationLatencyMicroseconds *uint64 `json:"averageIOOperationLatencyMicroseconds,omitempty"`
+
+	// maxIOOperationsPerSecondPerTerabyte to be supported or not.
+	// +nullable
+	maxIOOperationsPerSecondPerTerabyte *uint64 `json:"maxIOOperationsPerSecondPerTerabyte,omitempty"`
+
+	// storage tier to be used for pool or volume creation. eg(Platinum, Gold, Silver)
+	StorageTier StorageTier `json:"storageTier"`
+}
+
+type DataProtectionCapabilities struct {
+	// TODO: Decide the Fields
+}
+
+type StorageTier string
```

Comment:
As this is user-defined do we need to add a type here?
IMO this can be a string. As we don't have any validation on this. It is up to the storage admin how he/she want to name it. If it is a new type then if we want one more  StorageTier we need to add that in the code.
If we want to make it strict then good to have a type else we can go with string.

Reply:
I was under impression that the storage tier might be known. But it makes sense since the storage admin can name it and we can't do any validation. Will change it to string type.

Reply:
Yes,  it is a useer-defined string.